trieste/models/gpflow/models.py

```diff
@@ -38,7 +39,11 @@ class GaussianProcessRegression(GPflowPredictor, TrainableProbabilisticModel):
     """
 
     def __init__(
-        self, model: GPR | SGPR, optimizer: Optimizer | None = None, num_kernel_samples: int = 10
+        self,
+        model: GPR | SGPR,
+        optimizer: Optimizer | None = None,
+        num_kernel_samples: int = 10,
+        num_rff_features: int = 1000,
```

Comment:
choice of a sampler type seems to some extent independent from the model, or at least you could have more than 1 option, so these kind of arguments would be more appropriate at the sampler level...

Reply:
In the PR when we introduce more than one trajectory sampler type to Trieste, I will make sense to add support for this. Will leave as is for now

Reply:
shouldn't `num_rff_features` be optional? 
if its costly to initialise the sampler, we probably shouldn't create the sampler by default